### ISSUE TITLE
Arithmetic functions in the go template

### ISSUE THREAD

**Issue by micbar** [restic/calens#2]
## Description
Hi @fd0 

Thank you for this nice generator. It helps a lot! We just started to use the tool for our Github repos.

In the past, we were creating github compare links like
https://github.com/org/repo/compare/v1.0.0...v1.0.1

In the current implementation, we cannot use arithmetic functions in the go template on the index to get the version from the next / previous loop.

## Possible Solution
- We could add arithmetic helper functions to the go template (like `incr` and `decr`)
- We could use a library like Sprig (http://masterminds.github.io/sprig/)

Sprig would give you a lot of generic template fuctions. A benefit for the Calens project could specially be the Version compare functions http://masterminds.github.io/sprig/semver.html

Would you consider adding sprig as a dependency?

I would be happy to test!

**Comment by micbar** [comment]
@fd0 
I tested the sprig library https://github.com/micbar/calens/pull/1
It seems to work fine. 
We have a slight naming collision.

You introduced a helper function `wrap` which wraps text and indents it.
The sprig Library has also a `wrap` function.
We would need to resolve this naming conflict.

1) We could rename your wrap function to `wrapIndent` (which IMO would make more sense semantically). This would break compatibility with existing templates. Your function takes three arguments, the sprig function takes two.
2) We could rename the sprig function.

Please let me know what you prefer, i would then create a PR.

**Comment by fd0** [comment]
Hey, thanks for letting us know our little changelog generator is useful to others ! :)

I like your suggestion to add `sprig`, I wasn't aware that there even is a library for text templates. If `sprig` provides a `wrap` function I think it's valuable to use that.

Since the project isn't used by so many users and it's only 0.1.0 yet we could either do a breaking change (I'm fine with that personally) or use some Go magic with a variadic function like this:

```go
func wrapText(text string, width, indent int) (result string, err error) {
[...]
}

func wrap(text string, width, rest ...int) (result string, err error) {
    switch len(rest) {
         case 0:
             return sprig.Wrap(text, width) // need to figure out how to delegate this to sprig
         case 1:
             return wrapText(text, width, rest[0])
         default:
             return "", errors.New("unexpected number of parameters")
    }
}
```

That's ... ugly. And probably not worth it. At least when you have a template with a `wrap` function with three parameters running calens should fail, so users know they need to update their templates. Right?

What are your thoughts?

**Comment by micbar** [comment]
I personally like your `wrap` function very much because you can easily create valid markdown with indented list paragraphs using a single helper function.

So personally, i would like to keep it and rename it as proposed.

Let me do a quick test if the generator fails if the template provides too many arguments for a helper function.

**Comment by micbar** [comment]
It is as we suspected:

If you use the old templates with the new code, you will get something like this.

```
error executing template: template: :18:6: executing "" at <wrap>: wrong number of args for wrap: want 2 got 3 
```

So the users would know, that they need to update their template.

**Comment by micbar** [comment]
@fd0 See https://github.com/restic/calens/pull/3

**Comment by fd0** [comment]
cool, thanks!

**Comment by fd0** [comment]
Merged as #3
